dtfs-central-api/src/v1/controllers/utilisation-report-service/post-payment.controller/helpers.test.ts

```diff
@@ -124,6 +142,33 @@ describe('post-add-payment.controller helpers', () => {
       await expect(addPaymentToUtilisationReport(reportId, feeRecordIds, tfmUser, newPaymentDetailsWithGBPCurrency)).rejects.toThrow(InvalidPayloadError);
     });
 
+    it("throws the 'InvalidPayloadError' if the selected fee records have payments and do not match the fee records in the fee record payment group", async () => {
+      // Arrange
+      const selectedFeeRecordIds = [1, 2];
+      const paymentFeeRecordIds = [1, 2, 3];
+
+      const payments = [
+        PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+          .withFeeRecords(paymentFeeRecordIds.map((feeRecordId) => FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(feeRecordId).build()))
+          .build(),
+      ];
+      const feeRecordsWithGBPPaymentCurrency = paymentFeeRecordIds.map((feeRecordId) =>
+        FeeRecordEntityMockBuilder.forReport(utilisationReport)
+          .withId(feeRecordId)
+          .withPaymentCurrency('GBP')
+          .withStatus(FEE_RECORD_STATUS.DOES_NOT_MATCH)
+          .withPayments(payments)
+          .build(),
+      );
+      feeRecordFindByIdWithPaymentsAndFeeRecordsSpy.mockResolvedValue(feeRecordsWithGBPPaymentCurrency);
+
+      // Act / Assert
+      await expect(addPaymentToUtilisationReport(reportId, selectedFeeRecordIds, tfmUser, newPaymentDetails)).rejects.toThrow(InvalidPayloadError);
+
+      expect(feeRecordFindBySpy).toHaveBeenCalledWith({ id: In(selectedFeeRecordIds) });
+      expect(feeRecordFindByIdWithPaymentsAndFeeRecordsSpy).toHaveBeenCalledWith(selectedFeeRecordIds);
```

Comment:
Can we add `toHaveBeenCalledTimes` assertions please?

Reply:
Added to all spy assertions in this file thanks